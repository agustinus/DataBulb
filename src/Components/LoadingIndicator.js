import React from 'react';
import {StyleSheet, ActivityIndicator} from 'react-native';

export default class LoadingIndicator extends React.Component {
  render() {
    return (
      <ActivityIndicator
        animating={this.props.isLoading}
        style={this.props.isLoading ? styles.loadingContainer : {}}
        size="large"
      />
    );
  }
}

const styles = StyleSheet.create({
  loadingContainer: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
