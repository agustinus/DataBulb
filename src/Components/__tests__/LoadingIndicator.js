import React from 'react';
import LoadingIndicator from '../LoadingIndicator';

import ReactTestRenderer from 'react-test-renderer';

describe('LoadingIndicator', () => {
  test('renders correctly', () => {
    const tree = ReactTestRenderer.create(<LoadingIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly if it's loading", () => {
    const tree = ReactTestRenderer.create(
      <LoadingIndicator isLoading={true} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
