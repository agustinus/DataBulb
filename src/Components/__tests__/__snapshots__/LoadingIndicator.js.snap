// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="large"
  style={Object {}}
/>
`;

exports[`LoadingIndicator renders correctly if it's loading 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
